### PR TITLE
fix: Always get url mapping using helpers when using Spring Boot

### PR DESCRIPTION
**Pull request by Artur-**
The properties in application.properties use a relaxed mapping strategy in
Spring Boot so all variants of "url-mapping","urlmapping","urlMapping" are
mapped to the same value in VaadinConfigurationProperties.

Fixes https://github.com/vaadin/spring/issues/478

### CODE REVIEW

File: vaadin-spring/src/main/java/com/vaadin/flow/spring/RootMappedCondition.java [hunk 1]
```diff
@@ -43,8 +48,43 @@ public class RootMappedCondition implements Condition {
...
+     * This is a helper needed only when VaadinConfigurationProperties is not
+     * available for injection, e.g. in a condition.
+     *
+     * @param beanFactory
+     *            the bean factory
+     * @param environment
+     *            the application environment
+     * @return the url mapping or null if none is defined
+     */
+    public static String getUrlMapping(ListableBeanFactory beanFactory,
```
Comment (mshabarov) [inline vaadin-spring/src/main/java/com/vaadin/flow/spring/RootMappedCondition.java:70]: `beanFactory` argument is never used, can it be removed?

File: vaadin-spring/src/main/java/com/vaadin/flow/spring/VaadinConfigurationProperties.java [hunk 2]
```diff
@@ -31,6 +33,22 @@
 @ConfigurationProperties(prefix = "vaadin")
 public class VaadinConfigurationProperties {
 
+    /**
+     * Gets the url mapping using the given environment.
+     *
+     * This is a helper needed only when VaadinConfigurationProperties is not
```
Comment (mshabarov) [inline vaadin-spring/src/main/java/com/vaadin/flow/spring/VaadinConfigurationProperties.java:39]:
```suggestion
     * This is a helper method needed only when VaadinConfigurationProperties is not
```

File: vaadin-spring/src/main/java/com/vaadin/flow/spring/RootMappedCondition.java [hunk 3]
```diff
@@ -43,8 +48,43 @@ public class RootMappedCondition implements Condition {
...
+        String urlMapping = getUrlMapping(context.getBeanFactory(),
+                context.getEnvironment());
+        return isRootMapping(urlMapping);
+    }
+
+    /**
+     * Gets the url mapping in a way compatible with both plain Spring and
+     * Spring Boot.
+     *
+     * This is a helper needed only when VaadinConfigurationProperties is not
```
Comment (mshabarov) [inline vaadin-spring/src/main/java/com/vaadin/flow/spring/RootMappedCondition.java:59]: "This is a helper needed only when ..." looks like copy-paste from `VaadinConfigurationProperties`. Or should it be "RootMappedCondition" instead?

**Review by mshabarov — CHANGES_REQUESTED** [review]

File: vaadin-spring/src/main/java/com/vaadin/flow/spring/SpringVaadinServletService.java [hunk 4]
```diff
@@ -161,8 +161,7 @@ private String getFullPath(String path, String prefix) {
      * ClassNotFound or similar exceptions in plain Spring.
      */
     private boolean isSpringBootConfigured() {
-        Class<?> resourcesClass = resolveClass(
-                SPRING_BOOT_WEBPROPROPERTIES_CLASS);
+        Class<?> resourcesClass = resolveClass(SPRING_BOOT_WEBPROPERTIES_CLASS);
```
[the remark this answers is not in the document]
Comment (Artur-) [reply vaadin-spring/src/main/java/com/vaadin/flow/spring/SpringVaadinServletService.java:164]: How? We still need to resolve the class

File: vaadin-spring/src/main/java/com/vaadin/flow/spring/RootMappedCondition.java [hunk 5]
```diff
@@ -43,8 +48,43 @@ public class RootMappedCondition implements Condition {
...
+     * @param environment
+     *            the application environment
+     * @return the url mapping or null if none is defined
+     */
+    public static String getUrlMapping(ListableBeanFactory beanFactory,
+            Environment environment) {
+        if (SpringUtil.isSpringBoot()) {
+            try {
+                return (String) Class.forName(
+                        "com.vaadin.flow.spring.VaadinConfigurationProperties")
```
[the remark this answers is not in the document]
Comment (Artur-) [reply vaadin-spring/src/main/java/com/vaadin/flow/spring/RootMappedCondition.java:70]: `VaadinConfigurationProperties` is Spring Boot specific and `RootMappedCondition` is not

File: vaadin-spring/src/main/java/com/vaadin/flow/spring/RootMappedCondition.java (hunk 3 above)
Comment (Artur-) [reply vaadin-spring/src/main/java/com/vaadin/flow/spring/RootMappedCondition.java:59]: Fixed

File: vaadin-spring/src/main/java/com/vaadin/flow/spring/VaadinConfigurationProperties.java (hunk 2 above)
Comment (Artur-) [reply vaadin-spring/src/main/java/com/vaadin/flow/spring/VaadinConfigurationProperties.java:39]: Fixed

File: vaadin-spring/src/main/java/com/vaadin/flow/spring/SpringVaadinServletService.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (mshabarov) [reply vaadin-spring/src/main/java/com/vaadin/flow/spring/SpringVaadinServletService.java:164]: Ah, sorry, I overlooked `resourcesClass` is used further.

[the remark this answers is not in the document]
Comment (mshabarov) [reply vaadin-spring/src/main/java/com/vaadin/flow/spring/SpringVaadinServletService.java:164]: Maybe then make `resolveClass` public and reuse it?

File: vaadin-spring/src/main/java/com/vaadin/flow/spring/RootMappedCondition.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (mshabarov) [reply vaadin-spring/src/main/java/com/vaadin/flow/spring/RootMappedCondition.java:67]: Alright, so `VaadinConfigurationProperties` might be not present for "non-Boot" Spring apps. Thanks.

**Review by mshabarov — APPROVED** [review]

**Pull request merged**